src/Shared/UnitTests/MockLogger.cs

```diff
@@ -298,92 +276,91 @@ internal void LoggerEventHandler(object sender, BuildEventArgs eventArgs)
                 {
                     ExternalProjectStartedEvents.Add(args);
                 }
-                else if (eventArgs is ExternalProjectFinishedEventArgs)
+                else if (eventArgs is ExternalProjectFinishedEventArgs finishedEventArgs)
                 {
-                    ExternalProjectFinishedEvents.Add((ExternalProjectFinishedEventArgs) eventArgs);
+                    ExternalProjectFinishedEvents.Add(finishedEventArgs);
                 }
 
-                if (eventArgs is ProjectStartedEventArgs startedEventArgs)
-                {
-                    ProjectStartedEvents.Add(startedEventArgs);
-                }
-                else if (eventArgs is ProjectFinishedEventArgs)
+                switch (eventArgs)
```

Comment:
Good point. R# wasn't smart enough in this case to take it further.

---
In reply to: [197157928](https://github.com/Microsoft/msbuild/pull/3438#discussion_r197157928) [](ancestors = 197157928)

Reply:
Actually it was technically correct since there;s a missing "else" on this line. Semantically it's one big switch though.

---
In reply to: [197206639](https://github.com/Microsoft/msbuild/pull/3438#discussion_r197206639) [](ancestors = 197206639,197157928)